src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -704,10 +704,11 @@ def custom_forward(*inputs):
             if output_attentions:
                 all_attentions = all_attentions + (layer_outputs[1],)
 
-        hidden_states = self.layer_norm(hidden_states)
         if output_hidden_states:
             encoder_states = encoder_states + (hidden_states,)
 
+        hidden_states = self.layer_norm(hidden_states)
```

Comment:
I did this to align the pytorch and tensorflow models with the flax model, as the flax model uses the "Copied from" statement. The flax model ensures that everything returned under `hidden_states` is pre-LN, and that `last_hidden_state` is post-LN. I think it makes sense to have all of the `hidden_states` to look like how they did right after leaving their respective layers, as opposed to having one with an extra processing step behind it.

Reply:
(If I revert this, to make tests pass, I'll need to change the logic in the flax model to deviate from how it's done in BART)

Reply:
Thanks for the explanation - SGTM to unify them with this logic

Reply:
Sorry actually don't see why this would affect the `# Copied from...` statement? This only impacts (Flax)WhisperEncoder, which isn't copied from BART in any case?

Reply:
Ah you're right, reverted!

Reply:
Lovely thanks! 🙌